fairlearn/reductions/_exponentiated_gradient/_lagrangian.py

```diff
@@ -53,6 +55,7 @@ def __init__(self, X, sensitive_features, y, estimator, constraints, eps, B, opt
         self.lambdas = pd.DataFrame()
         self.n = self.X.shape[0]
         self.n_oracle_calls = 0
+        self.n_dummy_classifiers = 0
```

Comment:
Can we rename to something like: `n_oracle_calls_dummy_returned`. Below I think we should include the dummy calls among oracle calls (since they are technically the oracle calls).

Reply:
I see your point, but.... to my mind, the use of `n_oracle_calls` is for finding out the average time spent in the underlying classifier. That would suggest renaming `n_oracle_calls` to something like `n_oracle_calls_supplied_estimator`  but that's a bit unwieldly.

Is `n_oracle_calls` interesting from an algorithmic standpoint as well? I could perhaps add a get-only property calculated from `n_oracle_calls` and `n_oracle_calls_dummy_returned` to provide a way of getting at that mean.

Reply:
How about this:
* let's make `n_oracle_calls` count the number of times `_call_oracle` is called, so it's inclusive of both kinds of calls; this is an important characteristic of the optimization algorithm and a more fundamental property than how often the actual classifier gets called (the fact that some standard classifiers don't handle the trivial case is just a fluke that put us down this path)
* I think it's also useful to count `n_oracle_calls_dummy_returned`
* if you want to get an idea about the average time you're talking about, you can either do:
  * `sum(oracle_execution_times) / (n_oracle_calls - n_oracle_calls_dummy_returned)`; it's a slight overestimation, but for now I think it will do just fine
  * at some point in future, we can improve tracking of oracle calls and in addition to the times, we can also track which estimator was invoked, what was the incoming lambda vector, whether that oracle call was "saved in the cache" (as part of `best_h`) etc.

Reply:
Fixed as discussed.